compiler/qsc_passes/src/capabilitiesck.rs

```diff
@@ -0,0 +1,538 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod base;
+
+#[cfg(test)]
+mod adaptive;
+
+#[cfg(test)]
+mod adaptive_plus_integers;
+
+#[cfg(test)]
+pub mod common;
+
+use miette::Diagnostic;
+use qsc_data_structures::span::Span;
+use qsc_fir::{
+    fir::{
+        Block, BlockId, CallableImpl, Expr, ExprId, ExprKind, Global, Ident, Item, ItemKind,
+        LocalItemId, LocalVarId, Package, PackageLookup, Pat, PatId, PatKind, Res, SpecDecl,
+        SpecImpl, Stmt, StmtId, StmtKind,
+    },
+    ty::FunctorSetValue,
+    visit::Visitor,
+};
+use qsc_frontend::compile::RuntimeCapabilityFlags;
+use qsc_rca::{ComputeKind, ItemComputeProperties, PackageComputeProperties, RuntimeFeatureFlags};
+use thiserror::Error;
+
+#[derive(Clone, Debug, Diagnostic, Error)]
+pub enum Error {
+    #[error("cannot use a dynamic boolean value")]
```

Comment:
nitpick: I think it's either "bool" or "Boolean" since the latter is a person's last name and a proper noun while the former is the common shorthand. That said, Q# types split the difference and are all capitalized, leading to "Bool." Which ever we pick, we should be consistent between these errors and the ones used in other places, which might mean we should unify all errors to use "Bool" and "Int" etc. since that is why type inference will output.

Reply:
Updated.